### ISSUE TITLE
ASP.NET Core 3.0 TypeLoadExcpetion

### ISSUE THREAD

**Issue by gregkalapos** [elastic/apm-agent-dotnet#604]
#546 is about moving to 3.0, but I did already some manual testing (also with ASP.NET Core 3.0) and did not run into any issues.

Still, we got this reported:

```
System.TypeLoadException: Could not load type 'Microsoft.AspNetCore.Http.Internal.BufferingHelper' from assembly 'Microsoft.AspNetCore.Http, Version=3.0.0.0"
```

**Comment by tmksoft** [comment]
You can provoke the bug by setting the following in the appsettings.json:

  "ElasticApm": {
    "SecretToken": ....,
    "ServerUrls": ....,
    "ServiceName": ....,
    "CaptureHeaders": true,
    "CaptureBody": "all",
    "CaptureBodyContentTypes": "application/json"
  }

**Comment by gregkalapos** [comment]
## Impact 
Request body capturing does not work on ASP.NET Core 3.0 - The good thing: we do not crash the application, the exception is handled within the agent - so no drastic impact on the observed application - only the overhead of the exception which is handled. Furthermore the request body is not captured. The transaction and every other field is still captured fine if this bug happens.

More logs with callstack:

```
  {ApmMiddleware} Exception thrown while trying to fill request context for sampled transaction e8ea233fcbc04caa
System.TypeLoadException: Could not load type 'Microsoft.AspNetCore.Http.Internal.BufferingHelper' from assembly 'Microsoft.AspNetCore.Http, Version=3.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60'.
   at Elastic.Apm.AspNetCore.Extensions.HttpRequestExtensions.ExtractRequestBody(HttpRequest request, IApmLogger logger)
   at Elastic.Apm.AspNetCore.ApmMiddleware.GetRequestBody(HttpRequest request)
   at Elastic.Apm.AspNetCore.ApmMiddleware.FillSampledTransactionContextRequest(HttpContext context, Transaction transaction)
```

## Workaround 

Turn off request body capturing

## Why does this happen

We used the `HttpRequest.EnableRewind()` method to enable reading the request body, so we can capture it. Turns out that this method is gone from ASP.NET Core 3.0. 

## Solution 
Proposed solution in https://github.com/elastic/apm-agent-dotnet/pull/608

**Comment by vhatsura** [comment]
@gregkalapos, according to https://github.com/aspnet/AspNetCore/issues/12505, `EnableRewind` needs to be replaced by `EnableBuffering`.

Also, according to [blog post `Re-reading ASP.Net Core request bodies with EnableBuffering()`](https://devblogs.microsoft.com/aspnet/re-reading-asp-net-core-request-bodies-with-enablebuffering/), the extension method `EnableBuffering` was added in ASP.NET Core 2.1. So, it means we can change `EnableRewind` with `EnableBuffering`

**Comment by gregkalapos** [comment]
This issue was fixed in #608.

#546 is still open, so we'd like to invest more on 3.0.
